paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/PaymentMethodRowButton.kt

```diff
@@ -31,47 +50,158 @@ internal fun PaymentMethodRowButton(
+                    trailingContent = trailingContent,
+                    isCheckmarkRow = false
+                )
+            }
+        }
     }
+}
 
-    RowButton(
-        isEnabled = isEnabled,
-        isSelected = isSelected,
-        isClickable = isClickable,
-        onClick = onClick,
-        contentPaddingValues = PaddingValues(horizontal = 12.dp, vertical = contentPaddingValues),
-        verticalArrangement = Arrangement.Center,
-        modifier = modifier.fillMaxWidth().heightIn(min = 52.dp),
+@Composable
+internal fun PaymentMethodRowButtonRadio(
+    isEnabled: Boolean,
+    isSelected: Boolean,
+    isClickable: Boolean = isEnabled,
+    iconContent: @Composable RowScope.() -> Unit,
+    title: String,
+    subtitle: String?,
+    onClick: () -> Unit,
+    contentDescription: String? = null,
+    modifier: Modifier = Modifier.fillMaxWidth(),
+    trailingContent: (@Composable RowScope.() -> Unit)? = null,
+) {
+    Row(
+        modifier = modifier.selectable(
+            selected = isSelected,
+            enabled = isClickable,
+            onClick = onClick
+        ),
+        verticalAlignment = Alignment.CenterVertically
     ) {
-        Row(
-            horizontalArrangement = Arrangement.spacedBy(12.dp),
-            verticalAlignment = Alignment.CenterVertically,
+        RadioButton(
+            selected = isSelected,
+            onClick = onClick,
+            modifier = Modifier.align(Alignment.CenterVertically)
+        )
+        RowButtonContent(
+            contentPaddingValues = PaddingValues(
+                start = 0.dp,
+                end = 12.dp,
+                top = subtitle.paddingValues(),
+                bottom = subtitle.paddingValues()
+            ),
+            verticalArrangement = Arrangement.Center,
         ) {
-            iconContent()
-
-            TitleContent(
+            PaymentMethodContentExtras(
                 title = title,
                 subtitle = subtitle,
                 isEnabled = isEnabled,
-                contentDescription = contentDescription
+                contentDescription = contentDescription,
+                isSelected = isSelected,
+                iconContent = iconContent,
+                trailingContent = trailingContent,
+                isCheckmarkRow = false
             )
+        }
+    }
+}
 
-            if (trailingContent != null) {
-                Spacer(modifier = Modifier.weight(1f))
-                trailingContent()
-            }
+@Composable
+internal fun PaymentMethodRowButtonCheckmark(
+    isEnabled: Boolean,
+    isSelected: Boolean,
+    isClickable: Boolean = isEnabled,
+    iconContent: @Composable RowScope.() -> Unit,
+    title: String,
+    subtitle: String?,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier.fillMaxWidth(),
+    contentDescription: String? = null,
+    trailingContent: (@Composable RowScope.() -> Unit)? = null,
+) {
+    Row(
+        modifier = modifier.selectable(
```

Comment:
It looks like we do this for every variant. Can we move this logic into the `RowButtonContent`?